facebook/reason
# [atom-reason] Switch building from using CommonML to Jenga

The atom-reason dev will have to install jenga (don't forget to `opam update`!). As for the precise versions of deps jenga needs, I haven't listed them. There are like 50 and jenga accepts any version... The newest deps versions work though.

The jengaroot is copied over from my jengaboot repo. See the header comment for more details.

I've chosen to have a naive copy paste of the final artifacts from `_build/top/output.{js, map}` into the current `output_byte_debug_js/`. The reason is that, for one, this means we don't change any require paths in our current code, and for two, `ocamlbuild`/topkg isn't happy to see `.cmo`, `.cmi`, etc. alongside the artifacts we're sharing (`output.{js, map}`), so no matter how I tweaked the output directory name, reason's `make` complained. I _could_ adapt jengaroot.ml to do this but changing output paths is a bit tedious.

Known regressions: 
- Initial build might be potentially slower because jenga's constructing the db too I guess.
- Build output is like the ugly duckling of outputs. One day it'll become a pretty swan!

Test plan:
- Try `npm start`.
- Try `npm start` again. See infinite percent perf boost.
- Try `make`. Make sure nothing's broken.
- Try `opam upgrade reason`, assuming your reason is pinned locally to this repo.
